### PR TITLE
Pass circleVisibility to proposal discussions and common feed objects

### LINKED ISSUE #1696: Add visibility options for proposals
Similar to discussions, we should allow users to create proposals with restricted visibility to specific circles. 
But, for proposals we should always make sure that the proposal is visible to the eligible voters. 
Also if the creator of the proposal is not an eligible voter we should probably have an option to make the proposal visible to her.
Currently for all existing proposals the visibility should be public.

### PR DESCRIPTION
**Pull request by danielr18**

### CODE REVIEW

**Comment by danielr18** [comment]
@ariel-dao Is this how you would start to approach it?

I'm aware that I'm still missing:
- `circleVisibility` validation (check they include all eligible voters)
- `circleVisibility` when feed items are added to the common and parent common. What should we do in this case?

**Comment by ariel-dao** [comment]
@danielr18 I would check validation for circle visibility (voters) when it's added in to the discussion and bubble it to the proposal and feed item, this already seems complex and we'll likely have a ton of issues if we don't set up some unifier between the data of the entities, so I'd probably create a util for that

File: functions/src/proposals/business/proposals/getProposalTypeWeights.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { governanceDb } from "../../../governance/db"
+import { PROPOSALS } from "../../../shared/enums/governance/PROPOSALS"
+import { Weight } from "../../../shared/interfaces/governance/proposals/BaseProposal"
+import { CommonError } from "../../../util/errors"
+import { findGovCircleWithIndex } from "../../../util/helpers/findGovCircleWithIndex"
+
+export async function getProposalTypeWeights(
```
Comment (ariel-dao) [inline functions/src/proposals/business/proposals/getProposalTypeWeights.ts:7]: let's try to use arrow functions unless we need `this`

File: functions/src/proposals/business/proposals/getProposalTypeWeights.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    if (!proposalDefinition)
+      throw new CommonError("Proposal type is not defined in governance", {
+        type,
+        governance,
+      })
+
+    const proposalGovDefinition = proposalDefinition[args.circleId]
+
+    if (typeof proposalGovDefinition === "undefined")
```
Comment (ariel-dao) [inline functions/src/proposals/business/proposals/getProposalTypeWeights.ts:36]: use `isNil` from lodash

**Review by ariel-dao — APPROVED** [review]
Approved, check the comments please for future ref

**Pull request merged**